### PR TITLE
Kactus' image diffs

### LINKED ISSUE #2137: Image diff polish
@niik will do this 💃

### PR DESCRIPTION
**Pull request by joshaber**
Fixes #2137

Bring in Kactus' image diffs. A continuation of https://github.com/desktop/desktop/pull/2285.

Shout out to @mathieudutour for upstreaming this 🤘 

- [x] Use a resize observer.
- [x] Use less manual measurement?

screen shot 2017-08-11 at 1 11 30 pm
screen shot 2017-08-11 at 1 11 38 pm
screen shot 2017-08-11 at 1 11 46 pm
screen shot 2017-08-11 at 1 12 11 pm

### CODE REVIEW

**Comment by joshaber** [comment]
Alright, I'm gonna call this ready for review. There ares still some less than ideal bits around layout: we have to do some measurement and calculation in JS. I tried _really_ hard to move it entirely to CSS and got damn close but was foiled by `position: absolute`.

I don't think this is worth spending any more time on for now. It's still much better than what we have now 👍

File: app/src/ui/diff/image-diffs/difference-blend.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    const maxSize: React.CSSProperties = {
+      maxHeight: this.props.maxSize.height,
+      maxWidth: this.props.maxSize.width,
+    }
+
+    return (
+      <div className="image-diff-difference" ref={this.props.onContainerRef}>
+        <div className="sizing-container" ref={this.props.onContainerRef}>
```
Comment (shiftkey) [inline app/src/ui/diff/image-diffs/difference-blend.tsx]: This `onContainerRef` shouldn't be necessary if it's also attached to the outer div, right?

Comment (joshaber) [inline app/src/ui/diff/image-diffs/difference-blend.tsx]: Good catch 👍

**Comment by joshaber** [comment]
giphy-2

**Comment by xt0rted** [comment]
I just updated and I'm not seeing image diffs for SVGs. Is that intended behavior or should I file a bug about it?

**Comment by joshaber** [comment]
@xt0rted Are they being diffed as text instead?

**Comment by xt0rted** [comment]
@joshaber yea, my svgs are showing as text right now

**Comment by mathieudutour** [comment]
in Kactus I have a checkbox to toggle between visual diffs and text diffs (every text file has a visual diff, here it would apply only to svgs). Happy to send a PR if you want

**Comment by shiftkey** [comment]
@mathieudutour a PR would be :gem: - we might need some design input on it, but seeing the code would be a good starting point at least.

**Comment by mathieudutour** [comment]
Cool, will send one sometime this week. The design I have on kactus is pretty rough:
screen shot 2017-09-11 at 21 52 13

**Pull request merged**